### ISSUE TITLE
Write string array failed

### ISSUE THREAD

**Issue by tianxie** [jamesmudd/jhdf#656]
**Describe the bug**

Write string array failed if it contains non-ASCII characters.

**To Reproduce**

```java
        try (WritableHdfFile hdfFile = HdfFile.write(Paths.get("jhdf.hdf5"))) {
            WritableGroup group = hdfFile.putGroup("group");
            group.putDataset("strings1", new String[]{"hello"});  // success
            group.putDataset("strings2", new String[]{"你好"}); // failed
            String[][] strings = {{"你好"}, {"世界"}};
            group.putDataset("strings3", strings); // failed
        }
```

**Please complete the following information:**
- `jhdf` version: 0.8.4
- Java version: OpenJDK 64-Bit Server VM Temurin-17.0.9+9
- OS (Windows, Mac, Linux): Windows
- Stack trace if available
- `jhdf` logging if available

```
Exception in thread "main" java.nio.BufferOverflowException
	at java.base/java.nio.ByteBuffer.put(ByteBuffer.java:1007)
	at java.base/java.nio.HeapByteBuffer.put(HeapByteBuffer.java:243)
	at io.jhdf.object.datatype.StringData.writeArrayData(StringData.java:216)
	at io.jhdf.object.datatype.StringData.writeArrayData(StringData.java:210)
	at io.jhdf.object.datatype.StringData.writeData(StringData.java:194)
	at io.jhdf.WritableDatasetImpl.writeData(WritableDatasetImpl.java:243)
	at io.jhdf.WritableDatasetImpl.write(WritableDatasetImpl.java:225)
	at io.jhdf.WritableGroupImpl.write(WritableGroupImpl.java:181)
	at io.jhdf.WritableGroupImpl.write(WritableGroupImpl.java:181)
	at io.jhdf.WritableHdfFile.flush(WritableHdfFile.java:78)
	at io.jhdf.WritableHdfFile.close(WritableHdfFile.java:68)
	at com.example.Test.main(Test.java:18)
```

**Comment by jamesmudd** [comment]
Thanks for opening this issue, I will take a look ASAP.

**Comment by jamesmudd** [comment]
I have merged the fix for this will publish a release soon.

**Comment by jamesmudd** [comment]
Released in [v0.8.5](https://github.com/jamesmudd/jhdf/releases/tag/v0.8.5)

**Comment by tianxie** [comment]
Thank you so much for your quick fix.
